cat/ptx-proxy.cat

```diff
@@ -0,0 +1,81 @@
+PTX
+(* NVIDIA. Parallel Thread Execution ISA Version 7.5 *)
+
+(*******************)
+(*   Auxiliaries   *)
+(*******************)
+(* Events *)
+let strong-write = W & (RLX | REL)
+let strong-read = R & (RLX | ACQ)
+let strong-m = strong-write | strong-read
+let strong-operation = strong-write | strong-read | F
+
+(* Proxy *)
+let generic-proxy = GEN * GEN
+let surface-proxy = SUR * SUR
+let texture-proxy = TEX * TEX
+let constant-proxy = CON * CON
+let same-proxy = generic-proxy | surface-proxy | texture-proxy | constant-proxy
+
+(* Relations *)
+let sr-fence = ([strong-operation]; (sr & same-proxy); [F]) | ([F]; (sr & same-proxy); [strong-operation])
+let po-fence = ([strong-operation]; (po & same-proxy); [F]) | ([F]; (po & same-proxy); [strong-operation])
+let sr-m = ([strong-m]; (sr & loc & same-proxy); [strong-m])
+let po-m = ([M]; (po & loc & same-proxy); [M])
+let morally-strong = sr-fence | po-fence | sr-m | po-m
+(*******************)
+let fence-sc = [F & SC] ; sr; [F & SC]
+let rec observation = (morally-strong & rf) | (observation; rmw; observation)
+let release-pattern = ([W & REL]; po-loc?; [strong-write]) | ([F & ACQ_REL]; po; [strong-write])
+let acquire-pattern = ([strong-read]; po-loc?; [R & ACQ]) | ([strong-read]; po; [F & ACQ_REL])
+let sw = morally-strong & (release-pattern; observation+; acquire-pattern) | fence-sc
+let cause-base = (po*; (sw; po*)+) | po
+(*******************)
+(* Proxy-aware causality ordering *)
+let proxy-fence-ops = [F \ ALIAS]; (same-proxy & scta); [M]
+let proxy-ops-fence = [M]; (same-proxy & scta); [F \ ALIAS]
```

Comment:
what is `F \ ALIAS`?

is `scta` the same as `same_block_r`? Since this is a base relation, I like the short name but maybe add a comment (both in the cat file and when you define the name in `RelationNameRepository`)

Instead of defining `proxy-ops-fence`, replace all instances by `proxy-ops-fence^-1`

Reply:
- `F \ ALIAS` is changed to `F`. I previously added `GEN` tag to the `fence.proxy.alias`, and I thought this` proxy-fence-ops` should execlude the alias fence. As I tested we don't need the `GEN` tag to `fence.proxy.alias`, so the `GEN` in alias fence is also removed.
- the `scta` is the same as `same_block_r`. comments added to both cat file and the RelationNameRepository. I found there are more similar relations in vulkan like `ssg, swg, sqf, ssw`. I don't know would they all needed. But we might have to find a more general way of such relations.
- `proxy-ops-fence` replaced by` proxy-fence-ops^-1`

Reply:
I agree that we might need a more general solution when we add support for vulkan, but for the time being this should be enough

Reply:
The Tag is still there. I will remove it.